src/AsyncTCP.cpp

```diff
@@ -482,7 +482,12 @@ bool AsyncClient::connect(IPAddress ip, uint16_t port)
 #endif
 
     //Serial.printf("DEBUG: connect to %08x port %d using IP... ", ip_addr, port);
-    errno = 0; r = lwip_connect_r(sockfd, (struct sockaddr*)&serveraddr, sizeof(serveraddr));
+    errno = 0;
+#ifdef ESP_IDF_VERSION_MAJOR
```

Comment:
Could you please elaborate on which SDK combination defines ESP_IDF_VERSION_MAJOR but fails to define the _r versions of the functions? I use the arduino-esp32 adapter, which appears to define the macro (in file ~/.arduino15/packages/esp32/hardware/esp32/1.0.6/tools/sdk/include/esp32/esp_idf_version.h) and I have had no problems compiling the library with the _r functions.

Maybe you need to instead check for a minimum or maximum version of the version instead.

Reply:
This issue occurs when using an older version of arduino in conjunction with espidf, in the "arduino as esp-idf component" configuration. It doesn't affect the newer versions.

Here's a PlatformIO config(truncated):

```ini
[env]
platform = espressif32
framework = espidf, arduino
platform_packages = 
	framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#idf-release/v4.0

[env:esp32doit-devkit-v1]
lib_deps = 
	bblanchon/ArduinoJson@^6.17.3
	https://github.com/lorol/LITTLEFS.git
	https://github.com/Spirit532/AsyncTCPSock.git
	https://github.com/lorol/ESPAsyncWebServer
```